src/objects/datetime.rs

```diff
@@ -0,0 +1,305 @@
+use err::PyResult;
+use ffi::PyDateTimeAPI;
+use ffi::{PyDateTime_Check, PyDateTime_DateTimeType};
+use ffi::{
+    PyDateTime_DATE_GET_HOUR, PyDateTime_DATE_GET_MICROSECOND, PyDateTime_DATE_GET_MINUTE,
+    PyDateTime_DATE_GET_SECOND,
+};
+use ffi::{
+    PyDateTime_DELTA_GET_DAYS, PyDateTime_DELTA_GET_MICROSECONDS, PyDateTime_DELTA_GET_SECONDS,
+};
+use ffi::{
+    PyDateTime_DateTime_GET_DAY, PyDateTime_DateTime_GET_MONTH, PyDateTime_DateTime_GET_YEAR,
+};
+use ffi::{PyDateTime_DateType, PyDate_Check};
+use ffi::{PyDateTime_Date_GET_DAY, PyDateTime_Date_GET_MONTH, PyDateTime_Date_GET_YEAR};
+use ffi::{PyDateTime_DeltaType, PyDelta_Check};
+use ffi::{
+    PyDateTime_TIME_GET_HOUR, PyDateTime_TIME_GET_MICROSECOND, PyDateTime_TIME_GET_MINUTE,
+    PyDateTime_TIME_GET_SECOND,
+};
+use ffi::{PyDateTime_TZInfoType, PyTZInfo_Check};
+use ffi::{PyDateTime_TimeType, PyTime_Check};
+use object::PyObject;
+use std::os::raw::c_int;
+
+#[cfg(Py_3_6)]
+use ffi::{PyDateTime_DATE_GET_FOLD, PyDateTime_TIME_GET_FOLD};
+
+use instance::Py;
+use python::{Python, ToPyPointer};
+
+// Traits
+pub trait PyDateComponentAccess {
+    fn get_year(&self) -> u32;
+    fn get_month(&self) -> u32;
+    fn get_day(&self) -> u32;
+}
+
+pub trait PyDeltaComponentAccess {
+    fn get_days(&self) -> i32;
+    fn get_seconds(&self) -> i32;
+    fn get_microseconds(&self) -> i32;
+}
+
+pub trait PyTimeComponentAccess {
+    fn get_hour(&self) -> u32;
+    fn get_minute(&self) -> u32;
+    fn get_second(&self) -> u32;
+    fn get_microsecond(&self) -> u32;
+    #[cfg(Py_3_6)]
+    fn get_fold(&self) -> u8;
+}
+
+// datetime.date bindings
+pub struct PyDate(PyObject);
+pyobject_native_type!(PyDate, PyDateTime_DateType, PyDate_Check);
+
+impl PyDate {
+    pub fn new(py: Python, year: u32, month: u32, day: u32) -> PyResult<Py<PyDate>> {
+        let y = year as c_int;
+        let m = month as c_int;
+        let d = day as c_int;
+
+        unsafe {
+            let ptr = PyDateTimeAPI.Date_FromDate.unwrap()(y, m, d, PyDateTimeAPI.DateType);
+            Py::from_owned_ptr_or_err(py, ptr)
+        }
+    }
+
+    pub fn from_timestamp(py: Python, args: &PyObject) -> PyResult<Py<PyDate>> {
+        unsafe {
+            let ptr =
+                PyDateTimeAPI.Date_FromTimestamp.unwrap()(PyDateTimeAPI.DateType, args.as_ptr());
+            Py::from_owned_ptr_or_err(py, ptr)
+        }
+    }
+}
+
+impl PyDateComponentAccess for PyDate {
+    fn get_year(&self) -> u32 {
+        unsafe { PyDateTime_Date_GET_YEAR(self.as_ptr()) as u32 }
+    }
+
+    fn get_month(&self) -> u32 {
+        unsafe { PyDateTime_Date_GET_MONTH(self.as_ptr()) as u32 }
+    }
+
+    fn get_day(&self) -> u32 {
+        unsafe { PyDateTime_Date_GET_DAY(self.as_ptr()) as u32 }
+    }
+}
+
+// datetime.datetime bindings
```

Comment:
Could you give the save wrapper type real rust type doc comments with `///`? Maybe even copying a bit from https://docs.python.org/3/library/datetime.html#available-types. Like:
```
/// A datetime.datetime object, which is a combination of a date and a time
pub struct PyDateTime(PyObject);
```